lua/nordic/colors/init.lua

```diff
@@ -1,7 +1,13 @@
 local U = require 'nordic.utils'
-local C = require 'nordic.colors.nordic'
+local P = require 'nordic.colors.nordic'
+
+local C = {}
+
+local function build_palette()
+    -- make a new palette
+    C = vim.deepcopy(P)
+    C.build_palette = build_palette
```

Comment:
Are you sure this is correct?

Maybe just do

```lua
local function C.build_palette()
--...
end
```

Reply:
It is the easiest way of doing it.

You could use:
```lua
function C.build_palette()
    -- make a new palette
    C = vim.tbl_deep_extend('force', C, P)
```
but that leaves the last calls extended palette which makes things more complicated, especially with #124.

So I figured a deep copy would be the best way to go, but if `build_palette` is attached to the C table, then it will be overwritten before we can reassign it.

I think you could attach it to the P table, but that seemed less intuitive to me:
```lua
function P.build_palette()
    -- make a new palette
    C = vim.deepcopy(P)

    -- ...
end

P.build_palette()

return C
```

Reply:
I was actually just referring to this line:

```lua
    C.build_palette = build_palette
```

Reply:
Yah that's what I am talking about too

Reply:
I will add some better comments though one sec.